File: frontend/src/app/services/restaurant.service.ts
import {Injectable} from "@angular/core";
import {HttpClient, HttpParams, HttpResponse} from "@angular/common/http";
import {Globals} from '../global/globals';
import {Observable} from "rxjs";
import {RestaurantDto, RestaurantOpeningHoursDto} from "../dtos/restaurant";

@Injectable({
  providedIn: 'root'
})

export class RestaurantService {

  private restaurantBaseUri : string = this.globals.backendUri + "/restaurants";

  constructor(private httpClient: HttpClient, private globals: Globals) {}

  /**
   * Get the opening hours for the location.
   *
   * @return an Observable for the opening hours
   */
  getOpeningHours(): Observable<RestaurantOpeningHoursDto> {
    return this.httpClient.get<RestaurantOpeningHoursDto>(this.restaurantBaseUri + "/opening-hours");
  }

  /**
   * Get the restaurant information
   *
   * @return an Observable for the restaurant information
   */
  getRestaurantInfo(): Observable<RestaurantDto> {
    return this.httpClient.get<RestaurantDto>(this.restaurantBaseUri + "/restaurant");
  }
}

File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/service/RestaurantService.java
package at.ac.tuwien.sepr.groupphase.backend.service;

import at.ac.tuwien.sepr.groupphase.backend.endpoint.dto.RestaurantOpeningHoursDto;
import at.ac.tuwien.sepr.groupphase.backend.endpoint.dto.RestaurantDto;

/**
 * Service for operations on the restaurant (e.g. getting the opening hours).
 */
public interface RestaurantService {

    /**
     * Get the opening hours of the restaurant.
     *
     * @return the opening hours
     */
    RestaurantOpeningHoursDto getOpeningHours();

    /**
     * Get the restaurant information.
     *
     * @return the restaurant information
     */
    RestaurantDto getRestaurantInfo();
}
